man/man8/zfs.8

```diff
@@ -2054,9 +2054,11 @@ The reservation is kept equal to the volume's logical size to prevent unexpected
 behavior for consumers.
 Without the reservation, the volume could run out of space, resulting in
 undefined behavior or data corruption, depending on how the volume is used.
-These effects can also occur when the volume size is changed while it is in use
-.Pq particularly when shrinking the size .
-Extreme care should be used when adjusting the volume size.
+In particular, asynchronous write errors are not propagated to consumers without
+the use of O_DIRECT. These effects can also occur when the volume size is
```

Comment:
What layer of software is failing to propagate the error?  If it's ZFS, can we fix that bug?  If it's not ZFS, let's make that more clear in the messaging here.  We might also want to talk about how typical consumers handle the errors.  The examples in https://github.com/zfsonlinux/zfs/issues/2889 seem to be using `dd` to directly access the zvol (`/dev/zvol/...`).  I think it's more typical to access the zvol by creating a filesystem on top of it (ext4) or export it over iSCSI.  Do those those consumers have this problem?